packages/flutter/lib/src/widgets/focus_traversal.dart

```diff
@@ -4,21 +4,66 @@
 
 import 'dart:ui';
 
+import 'package:collection/collection.dart';
 import 'package:flutter/foundation.dart';
 import 'package:flutter/painting.dart';
 
 import 'actions.dart';
 import 'basic.dart';
 import 'editable_text.dart';
 import 'focus_manager.dart';
+import 'focus_scope.dart';
 import 'framework.dart';
 import 'scroll_position.dart';
 import 'scrollable.dart';
 
+// BuildContext/Element doesn't have a parent accessor, but it can be simulated
+// with visitAncestorElements. _getAncestor is needed because
+// context.getElementForInheritedWidgetOfExactType will return itself it happens
```

Comment:
Side note: I recently suffered from the same problem: https://github.com/flutter/flutter/pull/50355. Maybe it really shouldn't return itself...